pkg/reconciler/configmappropagation/configmappropagation.go

```diff
@@ -0,0 +1,343 @@
+		}
+	}
+	return cmp, err
+}
+
+func (r *Reconciler) reconcileConfigMap(ctx context.Context, cmp *v1alpha1.ConfigMapPropagation) error {
+	// List ConfigMaps in original namespace and create/update copy ConfigMap in current namespace.
+	var errs error
+	originalConfigMapList, err := r.configMapLister.ConfigMaps(cmp.Spec.OriginalNamespace).List(resources.ExpectedOriginalSelector(cmp.Spec.Selector))
+	if err != nil {
+		logging.FromContext(ctx).Error("Unable to get the ConfigMap list in original namespace", zap.Error(err))
+		return err
+	}
+	for _, configMap := range originalConfigMapList {
+		name := resources.MakeCopyConfigMapName(cmp.Name, configMap.Name)
+		source := types.NamespacedName{Namespace: cmp.Spec.OriginalNamespace, Name: configMap.Name}.String()
+		resourceVersion := configMap.ResourceVersion
+		var expectedStatus v1alpha1.ConfigMapPropagationStatusCopyConfigMap
+		// Variable "succeed" represents whether a create/update action is successful or not.
+		if err, succeed := r.createOrUpdateConfigMaps(ctx, cmp, configMap); err != nil {
+			logging.FromContext(ctx).Warn("Failed to propagate ConfigMap: ", zap.Error(err))
+			r.Recorder.Eventf(cmp, corev1.EventTypeWarning, configMapPropagationPropagateSingleConfigMapFailed,
+				fmt.Sprintf("Failed to propagate ConfigMap %v: %v", configMap.Name, err))
+			if errs == nil {
+				errs = fmt.Errorf("one or more ConfigMap propagation failed")
+			}
+			expectedStatus.SetCopyConfigMapStatus(name, source, copyConfigMap, "False", err.Error(), resourceVersion)
+		} else if !succeed {
+			// If there is no error, but the create/update action is not successful,
+			// this indicates the copy configmap's copy label is removed.
+			logging.FromContext(ctx).Debug("Stop propagating ConfigMap " + configMap.Name)
+			r.Recorder.Eventf(cmp, corev1.EventTypeNormal, configMapPropagationPropagateSingleConfigMapSucceed,
+				fmt.Sprintf("Stop propagating ConfigMap: %s", configMap.Name))
+			expectedStatus.SetCopyConfigMapStatus(name, source, stopConfigMap, "True",
+				`copy ConfigMap doesn't have "knative.dev/config-propagation:copy" label, stop propagating this ConfigMap`, resourceVersion)
+		} else {
+			logging.FromContext(ctx).Debug("Propagate ConfigMap " + configMap.Name + " succeed")
+			r.Recorder.Eventf(cmp, corev1.EventTypeNormal, configMapPropagationPropagateSingleConfigMapSucceed,
+				fmt.Sprintf("Propagate ConfigMap %v succeed", configMap.Name))
+			expectedStatus.SetCopyConfigMapStatus(name, source, copyConfigMap, "True", "", resourceVersion)
+		}
+		// Update current copy configmap's status.
+		cmp.Status.CopyConfigMaps = append(cmp.Status.CopyConfigMaps, expectedStatus)
+	}
+	// List ConfigMaps in current namespace and delete copy ConfigMap if the corresponding original ConfigMap no longer exists or no longer has the required label.
+	copyConfigMapList, err := r.configMapLister.ConfigMaps(cmp.Namespace).List(labels.Everything())
```

Comment:
Yeah, I listed everything because we used to check copy configmap with copy label removed by users, and returned error. Since we are no longer do that (currently using stop propagating for copy configmap without copy label), we can only return configmaps with copy label.